packages/quicktype-core/src/Messages.ts

```diff
@@ -286,6 +286,8 @@ export function messageError<Kind extends ErrorKinds>(
             valueString = value.message;
         } else if (typeof value !== "string") {
             valueString = JSON.stringify(value);
+        } else {
+            valueString = value;
```

Comment:
Essentially, none of the cases above actually handled string, with the last else-if cases handling "everything else than strings".